bravado/swagger_model.py

```diff
@@ -86,12 +92,40 @@ def load_spec(self, spec_url, base_url=None):
         :param base_url: TODO: need this?
         :returns: json spec in dict form
         """
-        spec_json = request(
+        response = request(
             self.http_client,
             spec_url,
             self.request_headers,
-        ).result().json()
-        return spec_json
+        ).result()
+
+        content_type = response.headers.get('content-type', '').lower()
+        if is_yaml(spec_url, content_type):
+            return self.load_yaml(response.text)
+        else:
+            return response.json()
+
+    def load_yaml(self, text):
+        """Load a YAML Swagger spec from the given string, transforming
```

Comment:
This suggests to me that your YAML spec is incorrect, we shouldn't need to cast these to string. Need to wrap in quotes.
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#responses-object-example